packages/streams/canboatjs.js

```diff
@@ -48,18 +48,22 @@ function CanboatJs (options) {
 require('util').inherits(CanboatJs, Transform)
 
 CanboatJs.prototype._transform = function (chunk, encoding, done) {
-  if (_.isObject(chunk) && chunk.fromFile) {
-    this.fromPgn.parse(chunk.data, (err, pgnData) => {
-      if (err) {
-        console.error(err)
-      } else {
-        pgnData.timestamp = new Date(Number(chunk.timestamp)).toISOString()
-        this.push(pgnData)
-        this.app.emit('N2KAnalyzerOut', pgnData)
+  try {
+    if (_.isObject(chunk) && chunk.fromFile) {
+      this.fromPgn.parse(chunk.data, (err, pgnData) => {
+        if (err) {
+          console.error(err)
+        } else {
+          pgnData.timestamp = new Date(Number(chunk.timestamp)).toISOString()
+          this.push(pgnData)
+          this.app.emit('N2KAnalyzerOut', pgnData)
       }
-    })
-  } else {
-    this.fromPgn.parse(chunk, this.handlePgnData)
+      })
+    } else {
+      this.fromPgn.parse(chunk, this.handlePgnData)
+    }
+  } catch ( error ) {
+    console.error(err)
```

Comment:
This will fail when failing...err vs error. TypeScript, anyone..

Reply:
ouch

Reply:
nice catch!

Reply:
fixed and actually tested the error condition this time